clients/drcachesim/docs/drcachesim.dox.in

```diff
@@ -1062,44 +1057,49 @@ A #DR_ISA_REGDEPS #instr_t contains the following information:
   instr_get_src().
 - ISA mode: is always #DR_ISA_REGDEPS.  This information can be obtained using
   instr_get_isa_mode().
-- encoding bytes: an array of bytes containing the #DR_ISA_REGDEPS #instr_t
+- Encoding bytes: an array of bytes containing the #DR_ISA_REGDEPS #instr_t
   encoding.  Note that this information is present only for decoded instructions
   (i.e., #instr_t generated by decode() or decode_from_copy()).  This information
   can be obtained using instr_get_raw_bits().
-- length: the length of the encoded instruction in bytes.  Note that this
+- Length: the length of the encoded instruction in bytes.  Note that this
   information is present only for decoded instructions (i.e., #instr_t generated by
-  decode() or decode_from_copy()).  This information can be obtained by accessing
-  the #instr_t length field.
+  decode() or decode_from_copy()).  This information can be obtained using instr_length().
+  Be aware that in Google Workload Traces the instruction fetch size of a
+  #dynamorio::drmemtrace::memref_t# and the instr_length() of the corresponding fetched
+  instruction do not match! For convenience reasons we kept the instruction fetch size to
+  be the same as the size of the original ISA instruction.
 
 Note that all routines that operate on #instr_t and #opnd_t are also supported for
-#DR_ISA_REGDEPS instructions.  However, querying information outside of those
-described above (e.g., the instruction opcode with instr_get_opcode()) will return
-the zeroed value set by instr_create() or instr_init() when the #instr_t was
-created (e.g., instr_get_opcode() would return OP_UNDECODED).
+#DR_ISA_REGDEPS instructions and their operands.  However, querying information outside
+of those described above (e.g., the instruction opcode with instr_get_opcode()) will
+return the zeroed value set by instr_create() or instr_init() when the #instr_t was
+created.
 
 On top of instructions and memory acceses, traces also have
 #dynamorio::drmemtrace::trace_marker_type_t markers.
 All markers of the original trace are present, except for:
-#dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_IDX
-#dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL
-#dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_TRACE_START
-#dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_TRACE_END
-#dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_FAILED, which have been removed.
+- #dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_IDX
+- #dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL
+- #dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_TRACE_START
+- #dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_TRACE_END
+- #dynamorio::drmemtrace::TRACE_MARKER_TYPE_SYSCALL_FAILED
```

Comment:
And signal number

Reply:
Added.
I also added `TRACE_MARKER_TYPE_UNCOMPLETED_INSTRUCTION`. Should I just not mention it since new traces might have it once we handle it?
Should handling it be part of public v3 instead?

Reply:
Maybe there's a chance we would release more traces w/ the same format except the UNCOMPLETED?  For now probably simplest to just not mention it since the version inside the traces should rule out that marker (ignoring the version mishap over the last month).

Reply:
Removed.